### PR TITLE
fix(internal): Limit interface name size to os limit

### LINKED ISSUE #1456: ``kraft net create`` fails if network name is too long
### Describe the bug

When trying to create a network, there is a name length limit that is generated by the ``kraftkit`` convention of naming interfaces. On linux, for instance, this limit is 15 characters.

A possible solution would be to have some internal mapping from the network name to some random interface name that adheres to the limit.

### Steps to reproduce

Run:
```bash
kraft net create a-somewhat-longer-name

E  bridge creation for a-somewhat-longer-name failed: numerical result out of range
```

### Expected behavior

For the network to be created

### Which architectures were you using or does this bug affect?

_No response_

### Which operating system were you using or does this bug affect?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [ ] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [ ] Tested your changes against relevant architectures and platforms;
  - [ ] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

GitHub-Fixes: #1456

### CODE REVIEW

File: internal/cli/kraft/net/create/create.go [hunk 1]
```diff
@@ -20,6 +20,13 @@ import (
 	"kraftkit.sh/machine/network"
 )
 
+const (
+	// defaultNetworkMaxSize is the maximum size of a network name on Linux.
+	// We subtract 2 to account for the network interface suffix.
+	// The value 16 is taken from the glibc headers.
+	defaultNetworkNameMaxSize = 15 - 2
```
Comment (LucaSeri) [inline internal/cli/kraft/net/create/create.go]: This is still a somewhat tight limit. We append at least 4 characters, possibly going into double digits or maybe more (`@if%d`). A safe limit could be obtained by looking at the mask size of the network and seeing the number of interfaces that we could have on that network, but this limit might be too loose for the average use case and prevent users from naming their network properly.

Maybe we could have this limit (maybe rather 15 - 4) and warn the users about not being able to fully use the address space, possibly erroring in a better way if we run out of names for interfaces?

Comment (craciunoiuc) [reply internal/cli/kraft/net/create/create.go]: Ah sorry I thought it was just a number, then I'll decrease to accomodate for `@if%d%d`

Comment (craciunoiuc) [reply internal/cli/kraft/net/create/create.go]: As for erroring in the code when out of bounds makes sense, though I can't fully point the place where this should happen. Maybe you can do this in a separate PR? 100 interfaces should be enough for now :D

File: internal/cli/kraft/net/create/create.go [hunk 2]
```diff
@@ -56,9 +63,13 @@ func NewCmd() *cobra.Command {
 	return cmd
 }
 
-func (opts *CreateOptions) Pre(cmd *cobra.Command, _ []string) error {
+func (opts *CreateOptions) Pre(cmd *cobra.Command, args []string) error {
 	opts.Driver = cmd.Flag("driver").Value.String()
 
+	if len(args[0]) > defaultNetworkNameMaxSize {
```
Comment (LucaSeri) [inline internal/cli/kraft/net/create/create.go]: Thanks @craciunoiuc!
One last thing, can you please move this to ``Run`` and also add a check ``len(args) > 0``? Some places including compose use this as an API directly :pray:

**Review by LucaSeri — CHANGES_REQUESTED** [review]

File: internal/cli/kraft/net/create/create.go (hunk 2 above)
Comment (craciunoiuc) [reply internal/cli/kraft/net/create/create.go]: sure thingy

**Review by LucaSeri — APPROVED** [review]
Thanks! Some tests are failing but logic looks good.

**Comment by craciunoiuc** [comment]
It seems we were not respecting that limit ourselves in the tests, will change now

**Comment by craciunoiuc** [comment]
@LucaSeri limited test if names and works now

File: internal/cli/kraft/net/create/create.go [hunk 3]
```diff
@@ -20,6 +20,13 @@ import (
 	"kraftkit.sh/machine/network"
 )
 
+const (
+	// defaultNetworkMaxSize is the maximum size of a network name on Linux.
+	// We subtract 5 to account for the network interface suffix '@if%d%d%d'.
+	// The value 16 is taken from the glibc headers.
+	defaultNetworkNameMaxSize = 15 - 5
```
Comment (nderjung) [inline internal/cli/kraft/net/create/create.go]: This should be internalised to the bridge network implementation, since it is specific to Linux.

Comment (craciunoiuc) [reply internal/cli/kraft/net/create/create.go]: done

Comment (nderjung) [reply internal/cli/kraft/net/create/create.go]: Perfect, thanks!

**Review by nderjung — APPROVED** [review]
Thanks!

**Pull request merged**